python/cudf/cudf/utils/utils.py

```diff
@@ -160,16 +162,41 @@ def cudf_dtype_from_pydata_dtype(dtype):
     return infer_dtype_from_object(dtype)
 
 
-def is_single_value(val):
+def is_scalar(val):
     return (
-        isinstance(val, str)
+        val is None
+        or isinstance(val, str)
         or isinstance(val, numbers.Number)
         or np.isscalar(val)
         or isinstance(val, pd.Timestamp)
         or isinstance(val, pd.Categorical)
     )
 
 
+def to_cudf_compatible_scalar(val, dtype=None):
+    """
+    Converts the value `val` to a numpy/Pandas scalar,
+    optionally casting to `dtype`.
+
+    If `val` is None, returns None.
+    """
+    if val is None:
+        return val
+
+    if not is_scalar(val):
+        raise ValueError(
+            f"Cannot convert value of type {type(val).__name__} "
+            " to cudf scalar"
+        )
+
+    val = pd.api.types.pandas_dtype(type(val)).type(val)
```

Comment:
This is minor, but if we reverse the order can't we avoid reflection when the dtype is known?
```python
if dtype is not None:
  return pd.api.types.pandas_dtype(dtype).type(val)
return pd.api.types.pandas_dtype(type(val)).type(val)
```

Reply:
This doesn't quite work for datetime types as `pd.api.types.pandas_dtype(dtype).type` loses the resolution information. I agree there's probably a more efficient way to do this - any ideas?

Reply:
No you're right, I don't see any way around that problem other than explicitly checking for 
and handling the datetime resolutions, which this way already does implicitly.